### PR TITLE
feat(medusa, core-flows): add v2 endpoints for create-customer and customers/me

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Add `create-customer` endpoint to api-v2
- Add `customers/me` endpoint to api-v2

**Why**
- In order to properly develop and test auth middlewares we need endpoints for creating-customers and an authenticated endpoint. This could have worked with users but since the customer module is coming along I used that

### CODE REVIEW

File: packages/core-flows/src/definition/customer/create-customer.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import { createWorkflow, WorkflowData } from "@medusajs/workflows-sdk"
+import { createStoreCustomerStep } from "../../handlers/customer"
+
+type WorkflowInput = { customersData: CreateCustomerDTO[] }
+
+export const createStoreCustomersWorkflowId = "create-store-customers"
+export const createStoreCustomersWorkflow = createWorkflow(
+  createStoreCustomersWorkflowId,
+  (input: WorkflowData<WorkflowInput>): WorkflowData<CustomerDTO[]> => {
+    return createStoreCustomerStep(input.customersData)
```
Comment (srindom) [inline packages/core-flows/src/definition/customer/create-customer.ts]: thought: there should be an additional step in this workflow along the lines of:

```js
  const [customer] = createStoreCustomerStep(input.customersData)
  associateCustomerWithAuthUser(input.authUserId, customer.id)
  return customer
```

File: packages/core-flows/src/definition/customer/create-customer.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { CreateCustomerDTO, CustomerDTO } from "@medusajs/types"
+import { createWorkflow, WorkflowData } from "@medusajs/workflows-sdk"
+import { createStoreCustomerStep } from "../../handlers/customer"
+
+type WorkflowInput = { customersData: CreateCustomerDTO[] }
+
+export const createStoreCustomersWorkflowId = "create-store-customers"
+export const createStoreCustomersWorkflow = createWorkflow(
```
Comment (srindom) [inline packages/core-flows/src/definition/customer/create-customer.ts]: would rename this to something that also signifies that there is an "account" creation

File: packages/core-flows/src/definition/customer/create-customer.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { CreateCustomerDTO, CustomerDTO } from "@medusajs/types"
+import { createWorkflow, WorkflowData } from "@medusajs/workflows-sdk"
+import { createStoreCustomerStep } from "../../handlers/customer"
+
+type WorkflowInput = { customersData: CreateCustomerDTO[] }
```
Comment (srindom) [inline packages/core-flows/src/definition/customer/create-customer.ts:5]:
```suggestion
type WorkflowInput = { customersData: CreateCustomerDTO, authUserId: string }
```

File: packages/medusa/src/api-v2/store/customers/route.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import { createStoreCustomersWorkflow } from "@medusajs/core-flows"
+import { CreateCustomerDTO } from "@medusajs/types"
+
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  const createCustomers = createStoreCustomersWorkflow(req.scope)
+  const customersData = [req.validatedBody as CreateCustomerDTO]
+
+  const { result } = await createCustomers.run({
+    input: { customersData },
+  })
```
Comment (srindom) [inline packages/medusa/src/api-v2/store/customers/route.ts]: thought: would do something like this

```
  const authUserId = // however the authUser.id is placed on req
  const createCustomerAccount = createCustomerAccountWorkflow(req.scope)
  const customersData = req.validatedBody as CreateCustomerDTO

  const { result } = await createCustomerAccount.run({
    input: { customersData, authUserId },
  })
```

File: packages/core-flows/src/definition/customer/create-customer.ts (hunk 1 above)
Comment (srindom) [inline packages/core-flows/src/definition/customer/create-customer.ts]: we can reuse the step to create customers both in admin and "store account creation"

File: packages/medusa/src/api-v2/store/customers/query-config.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  "groups",
+]
+export const allowedStoreCustomersRelations = [
+  ...defaultStoreCustomersRelations,
+]
+export const defaultStoreCustomersFields: (keyof CustomerDTO)[] = [
+  "id",
+  "email",
+  "default_billing_address_id",
+  "default_shipping_address_id",
```
Comment (srindom) [inline packages/medusa/src/api-v2/store/customers/query-config.ts:16]: these will be removed in one of my PRs

**Review by srindom — COMMENTED** [review]
Just a few preliminary comments but looks good!

**Comment by pKorsholm** [comment]
@srindom I agree with all the points, I omitted the auth-module part of this pr to establish a foundation to build the authentication flow on top of in a subsequent pr 😄 I don't mind "drafting" this and making one larger pr later, but I don't think the auth-integration here is essential and would prefer to defer the work to a subsequent pr

**Comment by srindom** [comment]
In that case, I'd suggest you add TODOs in all the places where you intend to add things in a follow-up PR :)

**Pull request closed without merging**